cmake/utils.cmake

```diff
@@ -50,8 +50,8 @@ endfunction(effcee_default_c_compile_options)
 function(effcee_default_compile_options TARGET)
   effcee_default_c_compile_options(${TARGET})
   if (NOT "${MSVC}")
-    # RE2's public header requires C++11.  So publicly required C++11
-    target_compile_options(${TARGET} PUBLIC -std=c++11)
+    # RE2's public header requires C++11.  So publicly require C++11 or higher
+    target_compile_features(re2 PUBLIC cxx_std_11)
```

Comment:
This requires CMake 3.8 or later.

https://cmake.org/cmake/help/latest/release/3.8.html?highlight=cxx_std_11#c-c

SPIRV-Tools etc already require CMake 3.17 (because that's what our team's CI bots have installed).
So this has to be paired with an update to minimum cmake version.